homeassistant/components/controller/vera.py

```diff
@@ -0,0 +1,732 @@
+        updates the entity state. """
+        self._device_data = device_data
+        self.update_ha_state()
+
+    @property
+    def name(self):
+        """ Get the mame of the Controller. """
+        return self._name
+
+    def get_temperature(self):
+        """ Get the temperature and units. """
+        current_temp = self._device_data.get('temperature')
+        vera_temp_units = self._config.get('temperature_units', TEMP_CELCIUS)
+
+        if vera_temp_units == 'F':
+            temperature_units = TEMP_FAHRENHEIT
+        else:
+            temperature_units = TEMP_CELCIUS
+
+        return self._hass.config.temperature(
+            current_temp,
+            temperature_units)
+
+    @property
+    def state(self):
+        return self._state
+
+    @property
+    def should_poll(self):
+        """
+        Polling is not required as state is updated from the parent
+        """
+        return False
+
+    @property
+    def is_trippable(self):
+        """ Returns true if the device supports a trippable state """
+        return True if 'tripped' in self._device_data.keys() else False
+
+    @property
+    def is_armable(self):
+        """ Returns true if the device supports a armable state """
+        return True if 'armed' in self._device_data.keys() else False
+
+    @property
+    def is_armedtripped(self):
+        """ Returns true if the device is armed and tripped """
+        return True if 'armedtripped' in self._device_data.keys() else False
+
+    @property
+    def has_battery(self):
+        """ Returns true if the device supports battery level """
+        return True if 'batterylevel' in self._device_data.keys() else False
+
+    @property
+    def has_temperature(self):
+        """ Returns true if the device supports temperature level """
+        return True if 'temperature' in self._device_data.keys() else False
+
+    @property
+    def has_lightlevel(self):
+        """ Returns true if the device supports light level """
+        return True if 'light' in self._device_data.keys() else False
+
+    @property
+    def has_humidity(self):
+        """ Returns true if the device has a humidity reading """
+        return True if 'humidity' in self._device_data.keys() else False
+
+    @property
+    def is_switchable(self):
+        """ Returns true if the device supports switching """
+        if (self.category_name == "On/Off Switch"
+                or self.category_name == "Switch"):
+            return True
+        else:
+            return False
+
+    @property
+    def is_dimmable(self):
+        """ Returns true if the device supports dimming """
+        if self.category_name == "Dimmable Switch":
+            return True
+        else:
+            return False
+
+    @property
+    def category_name(self):
+        """ Returns the vera category name """
+        return self._device_data.get('categoryName', 'None')
+
+    # pylint: disable=too-many-statements
+    def create_child_devices(self):
```

Comment:
I think it would make more sense if this method was broken up and put into `sensor/vera`, `switch/vera`, `light/vera`. The logic would then drill down to:

```
Vera component:
  if connected:
    if has_sensor_values:
        load sensor.vera
        fire vera sensor discovered
    if has_switch_values:
        load switch.vera
        fire vera switch discovered
    if has_light_values:
        load light.vera
        fire vera light discovered
```
The `has_light_values` would take into consideration which switches are to be treated as lights.